### PR TITLE
Speed fields

### LINKED ISSUE #455: Update fields is too slow
Speed up the update fields function which seems to be the limiting point in our simulations with few particles and big rasters

### LINKED ISSUE #694: Add pfe to output files

### PR DESCRIPTION
**Pull request by matthiasto**
- Make computation of ppr, pta, P, TA and pke optional within the calculation loop . Only compute them if they are required as output
- fix #694 
- fix #455

### CODE REVIEW

File: avaframe/com1DFA/DFAfunctionsCython.pyx [hunk 1]
```diff
@@ -1058,9 +1054,56 @@ def updateFieldsC(cfg, particles, dem, fields):
   return particles, fields
 
 
-@cython.boundscheck(False)
-@cython.wraparound(False)
-@cython.cdivision(True)
+cpdef double computePressure(double v, double rho):
+  """Compute pressure"""
```
Comment (awirb) [inline avaframe/com1DFA/DFAfunctionsCython.pyx:1058]: just for completeness, add docstring

**Review by awirb — CHANGES_REQUESTED** [review]

File: avaframe/com1DFA/com1DFA.py [hunk 2]
```diff
@@ -1077,30 +1077,57 @@ def initializeFields(cfg, dem, particles):
...
+    resTypesLast = list(set(resTypes + resTypesReport))
     # read dem header
     header = dem['header']
     ncols = header['ncols']
     nrows = header['nrows']
-    PFV = np.zeros((nrows, ncols))
-    PP = np.zeros((nrows, ncols))
-    FT = np.zeros((nrows, ncols))
-    TA = np.zeros((nrows, ncols))
     fields = {}
```
[the remark this answers is not in the document]
Comment (matthiasto) [reply avaframe/com1DFA/com1DFA.py:1092]: I mean the function itself is called like this but I can add a comment

File: avaframe/com1DFA/com1DFA.py [hunk 3]
```diff
@@ -1077,30 +1077,57 @@ def initializeFields(cfg, dem, particles):
...
+    # even if it is empty to run properly
+    if ('TA' in resTypesLast) or ('pta' in resTypesLast):
+        fields['pta'] = np.zeros((nrows, ncols))
+        fields['TA'] = np.zeros((nrows, ncols))
+        fields['computeTA'] = True
+        log.info('Computing Travel Angle')
+    else:
+        fields['pta'] = np.zeros((1, 1))
+        fields['TA'] = np.zeros((1, 1))
+        fields['computeTA'] = False
```
[the remark this answers is not in the document]
Comment (matthiasto) [reply avaframe/com1DFA/com1DFA.py:1111]: What do you mean? should I do one for pta and a different flag for TA?

File: avaframe/com1DFA/com1DFA.py [hunk 4]
```diff
@@ -2103,6 +2095,8 @@ def exportFields(cfg, Tsave, fieldsList, demOri, outDir, logName):
             resField = fieldsList[countTime][resType]
             if resType == 'ppr':
                 resField = resField * 0.001
+            if resType == 'pke':
+                resField = resField * 0.001
```
Comment (fso42) [inline avaframe/com1DFA/com1DFA.py:2099]: Add a quick comment:
```
# convert from Pa to kPa
# convert from J to kJ
```

**Review by fso42 — CHANGES_REQUESTED** [review]

File: avaframe/com1DFA/com1DFA.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (awirb) [reply avaframe/com1DFA/com1DFA.py:1105]: still no log.info('Computing travel angle')

**Review by awirb — CHANGES_REQUESTED** [review]

File: avaframe/out3Plot/plotUtils.py [hunk 5]
```diff
@@ -204,15 +211,20 @@
...
+cmapEnergy = {}
+cmapEnergy['cmap'] = cmapE
+cmapEnergy['colors'] = colorsE
+cmapEnergy['levels'] = levE
+
 colorMaps = {'ppr': cmapPres, 'pfv': cmapSpeed, 'pft': cmapThickness, 'P': cmapPres, 'FV': cmapSpeed,
              'FM': cmapThickness, 'Vx': cmapSpeed, 'Vy': cmapSpeed, 'Vz': cmapSpeed, 'FTV': cmapSpeed,
-             'FT': cmapThickness, 'prob': cmapProb, 'pta': cmapTravelAngle, 'TA': cmapTravelAngle}
+             'FT': cmapThickness, 'prob': cmapProb, 'pta': cmapTravelAngle, 'TA': cmapTravelAngle,
+             'pke': cmapEnergy}
```
[the remark this answers is not in the document]
Comment (matthiasto) [reply avaframe/out3Plot/plotUtils.py:227]: there is no KE field, I only keep track of the peak value. The KE can be computed from FM and FV

File: avaframe/com1DFA/com1DFA.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (matthiasto) [reply avaframe/com1DFA/com1DFA.py:1105]: It is in the if part, not in the else

**Review by awirb — APPROVED** [review]

**Pull request merged**